### PR TITLE
Auth test

### PR DESCRIPTION
**Pull request by Qlexio**
Issue #51
# Testing existing code

* Add function to settings.py to test captcha
* Change path redirection in authentication.views.py to make it work
* Testing login and redirections
  * Test addition of new user with new mail
  * On database with existing users, test addition of user with:
    * different mail of existing users
    * same mail of existing users
  * Add 2 users with same mail and test result
* Testing token sent by mail and good redirections with:
  * Valid token
  * Invalid token
  * Already used token
  * Timed out token

### CODE REVIEW

**Comment by JeffAbrahamson** [comment]
PR #71 and #72 refer to the same branch.  This will be overly confusing.

https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-requests

File: transport_nantes/authentication/tests.py [hunk 1]
```diff
@@ -1,3 +1,163 @@
...
+
+# TODO Test good printing of pages (first print, post, redirect) and also page needing token from mail
+# TODO Test adding of new user if account doesn't exist and if account already exists
+class UserTest(TestCase):
+
+    # Add new user and test site page shown
+    def test_add_new_user(self):
+
+        # Set site into Site table
+        site = Site.objects.get(id=1)
```
Comment (JeffAbrahamson) [inline transport_nantes/authentication/tests.py:22]: Where was this set?

**Comment by JeffAbrahamson** [comment]
At commit b1df174 : 
* `python manage.py test` fails: the site record is not staged to the testing db.
* The change to ` transport_nantes/transport_nantes/settings.py` isn't appropriate -- all development should not be subject to that constraint, only auth tests.  This should be set in the test's `setup()` or `setUpTestData()` methods.

**Comment by JeffAbrahamson** [comment]
At 6024f71:
* `python manage.py test` fails - site still isn't set, but there appear to be some other errors (EObjectDoesNotExist).
* The settings file is reverted, which is good, but it leaves the flash of change in its history.

**Comment by JeffAbrahamson** [comment]
I'm now closing this PR in favour of #72 .

File: transport_nantes/authentication/tests.py (hunk 1 above)
Comment (Qlexio) [reply transport_nantes/authentication/tests.py:22]: This changes the value "exemple.com" to "localhost:8000" in Site model. If not done get_current_site in send_activation function fails and all the test fails also

**Comment by Qlexio** [comment]
> At commit [b1df174](https://github.com/transport-nantes/tn_web/commit/b1df17494efc5d379c9e4d4149b96921b04f1744) :
> 
>     * `python manage.py test` fails: the site record is not staged to the testing db.

The test fails probably because I use 'py manage.py test authentication --settings=transport_nantes.settings_tn' command instead of 'python manage.py test'

**Pull request closed without merging**